int32 encoder_count
string encoder_direction
